services/service/server/server_linux.go

```diff
@@ -266,10 +273,20 @@ func (s *server) Action(ctx context.Context, req *pb.ActionRequest) (*pb.ActionR
 	// NB: delivery of a value on resultchan respects context cancellation, and will
 	// deliver a value of 'cancelled' if the ctx is cancelled by a client disconnect,
 	// so it's safe to do a simple recv.
-	result := <-resultChan
-	if result != operationResultDone {
-		return nil, status.Errorf(codes.Internal, "error performing action %v: %v", req.Action, result)
+	// Enable/disable don't use this method so we skip the channel (since it would hang)
+	// and instead force a reload which is what systemctl does when it enables/disables.
+	switch req.Action {
```

Comment:
nit/optional: I typically always add a `default`: case that errors, in case someone adds a new enum value which isn't handled (there is one above, so it would already, but in case that value also needs handling, it's nice to have it be explicit)

Reply:
True. Added